crates/chia-consensus/src/gen/conditions.rs

```diff
@@ -4668,50 +4744,104 @@ fn test_softfork_condition_failures(#[case] conditions: &str, #[case] expected_e
 
 #[cfg(test)]
 #[rstest]
-#[case(CREATE_PUZZLE_ANNOUNCEMENT, 1000, None)]
+#[case(CREATE_PUZZLE_ANNOUNCEMENT, 1000, 0, None)]
 #[case(
     CREATE_PUZZLE_ANNOUNCEMENT,
     1025,
+    0,
     Some(ErrorCode::TooManyAnnouncements)
 )]
 #[case(
     ASSERT_PUZZLE_ANNOUNCEMENT,
     1024,
+    0,
     Some(ErrorCode::AssertPuzzleAnnouncementFailed)
 )]
 #[case(
     ASSERT_PUZZLE_ANNOUNCEMENT,
     1025,
+    0,
+    Some(ErrorCode::TooManyAnnouncements)
+)]
+#[case(CREATE_COIN_ANNOUNCEMENT, 1000, 0, None)]
+#[case(
+    CREATE_COIN_ANNOUNCEMENT,
+    1025,
+    0,
     Some(ErrorCode::TooManyAnnouncements)
 )]
-#[case(CREATE_COIN_ANNOUNCEMENT, 1000, None)]
-#[case(CREATE_COIN_ANNOUNCEMENT, 1025, Some(ErrorCode::TooManyAnnouncements))]
 #[case(
     ASSERT_COIN_ANNOUNCEMENT,
     1024,
+    0,
     Some(ErrorCode::AssertCoinAnnouncementFailed)
 )]
-#[case(ASSERT_COIN_ANNOUNCEMENT, 1025, Some(ErrorCode::TooManyAnnouncements))]
+#[case(
+    ASSERT_COIN_ANNOUNCEMENT,
+    1025,
+    0,
+    Some(ErrorCode::TooManyAnnouncements)
+)]
 #[case(
     ASSERT_CONCURRENT_SPEND,
     1024,
+    0,
     Some(ErrorCode::AssertConcurrentSpendFailed)
 )]
-#[case(ASSERT_CONCURRENT_SPEND, 1025, Some(ErrorCode::TooManyAnnouncements))]
+#[case(
+    ASSERT_CONCURRENT_SPEND,
+    1025,
+    0,
+    Some(ErrorCode::TooManyAnnouncements)
+)]
 #[case(
     ASSERT_CONCURRENT_PUZZLE,
     1024,
+    0,
+    Some(ErrorCode::AssertConcurrentPuzzleFailed)
+)]
+#[case(
+    ASSERT_CONCURRENT_PUZZLE,
+    1025,
+    0,
+    Some(ErrorCode::TooManyAnnouncements)
+)]
+// new flag tests
+#[case(CREATE_PUZZLE_ANNOUNCEMENT, 1025, COST_CONDITIONS, None)]
+#[case(
+    ASSERT_PUZZLE_ANNOUNCEMENT,
+    1025,
+    COST_CONDITIONS,
+    Some(ErrorCode::AssertPuzzleAnnouncementFailed)
+)]
+#[case(CREATE_COIN_ANNOUNCEMENT, 1025, COST_CONDITIONS, None)]
+#[case(
+    ASSERT_COIN_ANNOUNCEMENT,
+    1025,
+    COST_CONDITIONS,
+    Some(ErrorCode::AssertCoinAnnouncementFailed)
+)]
+#[case(
+    ASSERT_CONCURRENT_SPEND,
+    1025,
+    COST_CONDITIONS,
+    Some(ErrorCode::AssertConcurrentSpendFailed)
+)]
+#[case(
+    ASSERT_CONCURRENT_PUZZLE,
+    1025,
+    COST_CONDITIONS,
     Some(ErrorCode::AssertConcurrentPuzzleFailed)
 )]
-#[case(ASSERT_CONCURRENT_PUZZLE, 1025, Some(ErrorCode::TooManyAnnouncements))]
 fn test_limit_announcements(
     #[case] cond: ConditionOpcode,
     #[case] count: i32,
+    #[case] flag: u32,
```

Comment:
in `rstest` you can also add an independent parameter by saying:
```
#[values(0, COST_CONDITIONS)]
```
Then every test is run in both configuration. You could check the result programatically.